src/main.js

```diff
@@ -269,12 +275,14 @@ module.exports = {
         const text = textEditor.getText()
 
         let rules = {}
-        if (textEditor.isModified() && Object.keys(ignoredRulesWhenModified).length > 0) {
-          rules = ignoredRulesWhenModified
-        }
-        if (textEditor.isModified() && ignoreFixableRulesWhileTyping) {
-          // Note that this list will only contain rules after the first lint job
-          rules = idsToIgnoredRules(helpers.rules.getFixableRules())
+        if (textEditor.isModified()) {
+          const ignoredRules = new Set()
```

Comment:
This seems a lot cleaner than the `forEach` loop.
```
const ignoredRules = new Set(ignoredRulesWhenModified)
```

How about this for this whole section?
```
const ignoredRules = [
  ...ignoredRulesWhenModified,
  ...(
    ignoreFixableRulesWhileTyping
      ? helpers.rules.getFixableRules()
      :  []
  )
]
const uniqueIgnoredRules = new Set(ignoredRules)` 
rules = idsToIgnoredRules(uniqueIgnoredRules)
```

Reply:
> `const ignoredRules = new Set(ignoredRulesWhenModified)`

Definitely, changed.

> How about this for this whole section?

It's clearer to me at least the way it is.